scripts/sanity_chk/sanitylib.py

```diff
@@ -2899,7 +2899,7 @@ def calc_one_elf_size(instance):
     def discard_report(self, filename):
 
         try:
-            if self.discards is None:
```

Comment:
your commit message does not tell what you are fixing, it just says what the code already shows. What is the problem with the check that you are fixing?

Reply:
Reworded the commit message and added more description to that. Thanks. I am trying to fix the dead code. Earlier self.discards is None wasn't actually checking for empty discards dictionary.

Reply:
I am not familiar with this code but isn't an empty discard list a valid case ? Shouldn't you just ignore it instead of raise an error ? Correct if I'm wrong but the current code will just generate a file with no contents but the headers right ? Is it even a problem ?

Reply:
So discard_report function should only run when there is something in the discard dictionary. Otherwise the purpose of the discard report would be lost , isnt it? Therefore this function needs to raise the error if the apply_filter() function hasn't run before its execution.